src/Map.js

```diff
@@ -36,6 +36,7 @@ export default class Map {
   }
 
   constructor(options) {
+    const mapCanvas = document.getElementById('map-canvas')
```

Comment:
This line coupled the Map with the client who uses it. As we do in the mount, the Map doesn't know who it would be mounted to, that's the responsibility of the client.
https://github.com/Greenstand/treetracker-web-map-core/blob/402b70d56f0ba54b1f417d8f0c66ce2ceb49a79f/src/Map.js#L1256-L1258
Do you think we can update the width/height in the mount method?

Reply:
> @SuokN great work, I had a plan to do it in the server side: https://github.com/Greenstand/treetracker-query-api/blob/main/docs/api/spec/query-api.yaml#L841

It is great, i think it will work better.

Where can I look at the algorithm, that calculates clusters location? I am wondering, because of the case for example planters/7704, when we have one cluster with 2K trees and, then after zooming, the two clusters on the right side of the 2k center and we can't see them, and arrow does not work right.

Reply:
> This line coupled the Map with the client who uses it. As we do in the mount, the Map doesn't know who it would be mounted to, that's the responsibility of the client.
> 
> https://github.com/Greenstand/treetracker-web-map-core/blob/402b70d56f0ba54b1f417d8f0c66ce2ceb49a79f/src/Map.js#L1256-L1258
> 
> Do you think we can update the width/height in the mount method?

I think it good idea, i will try it. Thank you.

Reply:
The algorithm on the server side is an undecided thing, but the client side, we use this: 
https://github.com/Greenstand/treetracker-web-map-core/blob/402b70d56f0ba54b1f417d8f0c66ce2ceb49a79f/src/mapTools.js#L69-L141

basically, we fetch all the clusters (at a specified zoom level) and try to draw a rectangle to contain them all.

Reply:
> The algorithm on the server side is an undecided thing, but the client side, we use this:
> 
> https://github.com/Greenstand/treetracker-web-map-core/blob/402b70d56f0ba54b1f417d8f0c66ce2ceb49a79f/src/mapTools.js#L69-L141
> 
> basically, we fetch all the clusters (at a specified zoom level) and try to draw a rectangle to contain them all.
The client part I've seen. I mean how is the server calculate quantities of the clusters and location of this clusters here
url = `${
        this.apiServerUrl
      }trees?clusterRadius=${Map._getClusterRadius(
        radius,
      )}&zoom_level=${radius}&${this._getFilterParameters()}`